packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -203,6 +211,245 @@ describe('BelongsTo relation', () => {
+    const cartItem = await hasManyThroughRepo.create(
+      {
+        description: 'an item hasManyThrough',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const persistedItem = await cartItemRepo.findById(cartItem.id);
+    const persistedLink = await CustomerCartItemLinkRepo.find();
+    expect(cartItem).to.deepEqual(persistedItem);
+    expect(persistedLink).have.length(1);
+    const expected = {
+      id: 99,
+      customerId: existingCustomerId,
+      itemId: cartItem.id,
+    };
+    expect(toJSON(persistedLink[0])).to.deepEqual(toJSON(expected));
+  });
+
+  it('finds an instance via through model', async () => {
+    const item = await hasManyThroughRepo.create(
+      {
+        description: 'an item hasManyThrough',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const notMyItem = await cartItemRepo.create({
+      description: "someone else's item desc",
+    });
+
+    const items = await hasManyThroughRepo.find();
+
+    expect(items).to.not.containEql(notMyItem);
+    expect(items).to.deepEqual([item]);
+  });
+
+  it('finds instances via through models', async () => {
+    const item1 = await hasManyThroughRepo.create(
+      {
+        description: 'group 1',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const item2 = await hasManyThroughRepo.create(
+      {
+        description: 'group 2',
+      },
+      {
+        throughData: {id: 98},
+      },
+    );
+    const items = await hasManyThroughRepo.find();
+
+    expect(items).have.length(2);
+    expect(items).to.deepEqual([item1, item2]);
+    const group1 = await hasManyThroughRepo.find({
+      where: {description: 'group 1'},
+    });
+    expect(group1).to.deepEqual([item1]);
+  });
+
+  it('deletes an instance then delete the through model', async () => {
+    await hasManyThroughRepo.create(
+      {
+        description: 'customer 1',
+      },
+      {
+        throughData: {id: 98},
+      },
+    );
+    const anotherHasManyThroughRepo = hasManyThroughFactory(
+      existingCustomerId + 1,
+    );
+    const item2 = await anotherHasManyThroughRepo.create(
+      {
+        description: 'customer 2',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    let items = await cartItemRepo.find();
+    let links = await CustomerCartItemLinkRepo.find();
+
+    expect(items).have.length(2);
+    expect(links).have.length(2);
+
+    await hasManyThroughRepo.delete();
+    items = await cartItemRepo.find();
+    links = await CustomerCartItemLinkRepo.find();
+
+    expect(items).have.length(1);
+    expect(links).have.length(1);
+    expect(items).to.deepEqual([item2]);
+    expect(links[0]).has.property('itemId', item2.id);
+    expect(links[0]).has.property('customerId', existingCustomerId + 1);
+  });
+
+  it('through models get deleted once the related target gets deleted', async () => {
```

Comment:
`through model get deleted once the related target gets deleted` -> `deletes through model when corresponding target gets deleted`